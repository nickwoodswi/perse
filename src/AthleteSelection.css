.assignment-type{
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: center;
    width: 100%
}
    .athlete-selector{
        width: 100%;
        height: 70px;
    }
    .assignment-type input{
        width: 100%;
        height: 65px;
        margin: 0px;
    }
.assignment-type-toggle{
    background-color: yellow;
    display: flex;
    text-align: center;
    align-items: center;
    width: 120px;
    height: 70px;
    margin-left: 15px;
    margin-top: 5px;
}