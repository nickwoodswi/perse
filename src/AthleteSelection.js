import React, { Component } from 'react'
import './AthleteSelection.css'

class AthleteSelection extends Component {
    static defaultProps = {
        type: 'select',
        selectorOptions: []
      }
    render() {
        if (this.props.type === 'select') {
            return(
                <>
                <div className="assignment-type">
                    <label htmlFor="athlete-selector">Select Athlete For Assignment</label>
                    <select id="athlete-selector" className="athlete-selector" 
                        onChange={e => this.props.define(e.target.value, e.target.options[e.target.selectedIndex].attributes.getNamedItem('firstname').nodeValue, e.target.options[e.target.selectedIndex].attributes.getNamedItem('lastname').nodeValue )}>
                        <option>SELECT ATHLETE</option>
                        {this.props.selectorOptions.map((option, idx) => {
                            return(
                                <option key={idx} firstname={option.first_name} lastname={option.last_name} value={option.athletes_id}>{option.first_name} {option.last_name}</option>
                            )
                        })}
                    </select>
                    OR
                    <div 
                        className="assignment-type-toggle"
                        onClick={e => this.props.changeAthleteAssignmentType('create')}>+ADD NEW ATHLETE</div>
                </div>
                </>
            )
        }
        if (this.props.type === 'create') {
            return(
                <>
                <div className="assignment-type">
                    <label htmlFor="assignment-name-input">Create Athlete First Name</label>
                    <input 
                        onChange={e => this.props.define(this.props.id, e.target.value, this.props.lastName)}
                        id="assignment-name-input" 
                        type="text" 
                        value={this.props.firstName} 
                        placeholder="First Name" />
                    <label htmlFor="assignment-name-input">Create Athlete Last Name</label>
                    <input 
                        onChange={e => this.props.define(this.props.id, this.props.firstName, e.target.value)}
                        id="assignment-name-input" 
                        type="text" 
                        value={this.props.lastName} 
                        placeholder="Last Name" />
                    OR
                        <div 
                            className="assignment-type-toggle"
                            onClick={e => this.props.changeAthleteAssignmentType('select')}>SELECT ATHLETE</div>
                </div>
                </>
            )
        }
    }
}

export default AthleteSelection