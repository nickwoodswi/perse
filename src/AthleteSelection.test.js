import React from 'react';
import { render } from '@testing-library/react';
import AthleteSelection from './AthleteSelection';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders AthleteSelection component', () => {
    render(<MemoryRouter><ArtistForm /></MemoryRouter>);
});

it('AthleteSelection renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><AthleteSelection /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});