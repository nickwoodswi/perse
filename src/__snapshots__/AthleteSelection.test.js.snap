// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AthleteSelection renders the UI as expected 1`] = `
<div
  className="assignment-type"
>
  <select
    className="athlete-selector"
    onChange={[Function]}
  >
    <option>
      SELECT ATHLETE
    </option>
  </select>
  OR
  <div
    className="assignment-type-toggle"
    onClick={[Function]}
  >
    +ADD NEW ATHLETE
  </div>
</div>
`;
